build.rs

```diff
@@ -1,116 +1,76 @@
-            sdk_tar_file.to_str().unwrap(),
-            "-C",
-            sdk_output_dir.to_str().unwrap(),
-        ];
-        Command::new("tar").args(args).status().unwrap();
+    if env_var("DOCS_RS").is_some() {
+        // Skip linking and bindgen when building docs as docs.rs won't have the
+        // dependency present and can't download it.
+        return;
     }
 
-    #[cfg(target_arch = "x86")]
-    let lib_path = sdk_output_dir.join("lib").join("x86");
-    #[cfg(target_arch = "x86_64")]
-    let lib_path = sdk_output_dir.join("lib").join("x64");
-    #[cfg(target_arch = "arm")]
-    let lib_path = sdk_output_dir.join("lib").join("arm32");
-    #[cfg(target_arch = "aarch64")]
-    let lib_path = sdk_output_dir.join("lib").join("arm64");
-
-    let mut inc_arg = String::from("-I");
-    inc_arg.push_str(
-        sdk_output_dir
-            .join("include")
-            .join("c_api")
-            .to_str()
-            .unwrap(),
-    );
-
-    println!("cargo:rustc-link-search=native={}", lib_path.display());
-    println!("cargo:rustc-link-lib=dylib=Microsoft.CognitiveServices.Speech.core");
+    let sdk_dir = if let Some(sdk_dir) = env_var("MS_COG_SVC_SPEECH_SDK_DIR") {
+        // If the user has specified an SDK directory then use that and don't
+        // try and download another one.
+        PathBuf::from(sdk_dir)
+    } else {
+        let download_dir = {
+            let mut dir = PathBuf::from(std::env::var_os("OUT_DIR").unwrap());
+            dir.push("sdk_output");
+            dir
+        };
+        // Don't bother re-downloading the SDK if it looks like it's already
+        // there, unless it's been explicitly requested.
+        if env_var("MS_COG_SVC_SPEECH_RENEW_SDK").is_some()
+            || !download_dir.exists()
+            || fs::read_dir(&download_dir).unwrap().next().is_none()
+        {
+            fs::create_dir_all(&download_dir).unwrap();
+            download_sdk(&download_dir);
+        }
+        // Point the SDK dir at the thing that a user is most likely to point at
+        // if they download it themselves.
+        match TargetOs::get() {
+            TargetOs::Linux => download_dir.join(format!("SpeechSDK-Linux-{}", SPEECH_SDK_VERSION)),
+            TargetOs::MacOS => download_dir.join("MicrosoftCognitiveServicesSpeech.xcframework"),
+            TargetOs::Windows => download_dir,
+        }
+    };
 
-    let skip_bindgen = env::var("MS_COG_SVC_SPEECH_SKIP_BINDGEN")
-        .map(|v| v == "1")
-        .unwrap_or(false);
+    let link_params = link_params(&sdk_dir);
+    println!("cargo:rustc-link-search={}", link_params.search_dir);
+    println!("cargo:rustc-link-lib={}", link_params.lib_arg);
 
-    if skip_bindgen {
+    if env_var("MS_COG_SVC_SPEECH_SKIP_BINDGEN").is_some() {
         return;
     }
+    println!("cargo::rustc-cfg=bindgen");
+
+    let include_dir = match TargetOs::get() {
+        TargetOs::Linux | TargetOs::Windows => sdk_dir.join("include/c_api"),
```

Comment:
For Windows, the NuGet package has different dir structures than the Linux SDK, `include_dir` should be `sdk_dir.join("build/native/include/c_api")`

Reply:
Oh, thanks, that's really sad because I can only source the nuget package securely so I was going to use it for Linux too. I guess I'm going to have to add `MS_COG_SVC_SPEECH_LIB_DIR` and `MS_COG_SVC_SPEECH_INCLUDE_DIR` separately.